pkg/controller/operators/catalog/operator.go

```diff
@@ -142,7 +143,7 @@ func NewOperator(ctx context.Context, kubeconfigPath string, clock utilclock.Clo
 		clientAttenuator:         scoped.NewClientAttenuator(logger, config, opClient, crClient),
 	}
 	op.sources = grpc.NewSourceStore(logger, 10*time.Second, 10*time.Minute, op.syncSourceState)
-	op.reconciler = reconciler.NewRegistryReconcilerFactory(lister, opClient, configmapRegistryImage, op.now)
+	op.reconciler = reconciler.NewRegistryReconcilerFactory(lister, opClient, configmapRegistryImage, op.now, op.logger.IsLevelEnabled(logrus.DebugLevel))
```

Comment:
I would lean towards just passing in a logrus.Entry instance instead of a debug boolean

Reply:
Is the reason why so you can be able to have access to the operator logger in places that aren't currently possible?

Reply:
I've done so now - at least to the level of generating the pod spec.